vendor/github.com/stripe/goforit/flags.go

```diff
@@ -0,0 +1,243 @@
+
+func init() {
+	stats, _ = statsd.New(statsdAddress)
+}
+
+const DefaultInterval = 30 * time.Second
+
+type Backend interface {
+	Refresh() (map[string]Flag, error)
+}
+
+type csvFileBackend struct {
+	filename string
+}
+
+type jsonFileBackend struct {
+	filename string
+}
+
+func readFile(file string, backend string, parse func(io.Reader) (map[string]Flag, error)) (map[string]Flag, error) {
+	var checkStatus statsd.ServiceCheckStatus
+	defer func() {
+		stats.SimpleServiceCheck("goforit.refreshFlags."+backend+"FileBackend.present", checkStatus)
+	}()
+	f, err := os.Open(file)
+	if err != nil {
+		checkStatus = statsd.Warn
+		return nil, err
+	}
+	defer f.Close()
+	return parse(f)
+}
+
+func (b jsonFileBackend) Refresh() (map[string]Flag, error) {
+	return readFile(b.filename, "json", parseFlagsJSON)
+}
+
+func (b csvFileBackend) Refresh() (map[string]Flag, error) {
+	return readFile(b.filename, "csv", parseFlagsCSV)
+}
+
+type Flag struct {
+	Name string
+	Rate float64
+}
+
+type JSONFormat struct {
+	Flags []Flag `json:"flags"`
+}
+
+var flags = map[string]Flag{}
+var flagsMtx = sync.RWMutex{}
+
+// Enabled returns a boolean indicating
+// whether or not the flag should be considered
+// enabled. It returns false if no flag with the specified
+// name is found
+func Enabled(ctx context.Context, name string) (enabled bool) {
+	defer func() {
+		var gauge float64
+		if enabled {
+			gauge = 1
+		}
+		stats.Gauge("goforit.flags.enabled", gauge, []string{fmt.Sprintf("flag:%s", name)}, .1)
+	}()
+
+	// Check for an override.
+	if ctx != nil {
+		if ov, ok := ctx.Value(overrideContextKey).(overrides); ok {
+			if enabled, ok = ov[name]; ok {
+				return
+			}
+		}
+	}
+
+	flagsMtx.RLock()
+	defer flagsMtx.RUnlock()
+	if flags == nil {
+		enabled = false
+		return
+	}
+	flag := flags[name]
+
+	// equality should be strict
+	// because Float64() can return 0
+	if f := rand.Float64(); f < flag.Rate {
+		enabled = true
+		return
+	}
+	enabled = false
+	return
+}
+
+func flagsToMap(flags []Flag) map[string]Flag {
+	flagsMap := map[string]Flag{}
+	for _, flag := range flags {
+		flagsMap[flag.Name] = Flag{Name: flag.Name, Rate: flag.Rate}
+	}
+	return flagsMap
+}
+
+func parseFlagsCSV(r io.Reader) (map[string]Flag, error) {
+	// every row is guaranteed to have 2 fields
+	const FieldsPerRecord = 2
+
+	cr := csv.NewReader(r)
+	cr.FieldsPerRecord = FieldsPerRecord
+	cr.TrimLeadingSpace = true
+
+	rows, err := cr.ReadAll()
+	if err != nil {
+		return nil, err
+	}
+
+	flags := map[string]Flag{}
+	for _, row := range rows {
+		name := row[0]
+
+		rate, err := strconv.ParseFloat(row[1], 64)
+		if err != nil {
+			// TODO also track somehow
+			rate = 0
+		}
+
+		flags[name] = Flag{Name: name, Rate: rate}
+	}
+	return flags, nil
+}
+
+func parseFlagsJSON(r io.Reader) (map[string]Flag, error) {
+	dec := json.NewDecoder(r)
+	var v JSONFormat
+	err := dec.Decode(&v)
+	if err != nil {
+		fmt.Printf("WTF? %s\n", err)
+		return nil, err
+	}
+	spew.Dump(v.Flags)
```

Comment:
It looks like we accidentally vendored this in 😲 

Can you vendor in a clean master?